### ISSUE TITLE
Google Backup Sync Stuck When Adguard 4.1 for Android (Enabled)

### ISSUE THREAD

**Issue by Birraque** [AdguardTeam/AdguardForAndroid#4957]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.1 (Beta)

### Environment

```markdown
- OS: Android 13 (MIUI 14)
- Device: Poco F3 (Redmi K40); Model: M2012K11AG; alioth
- Firmware: MIUI 14.0.7.0.TKHCNMX (Stable)
```

### Root access

- [X] Yes, I have it.

### What filters do you have enabled?

AdGuard Base filter, AdGuard Mobile Ads filter, AdGuard Tracking Protection filter, AdGuard Social Media filter, AdGuard Cookie Notices filter, AdGuard Popups filter, AdGuard Mobile App Banners filter, AdGuard Widgets filter, AdGuard Other Annoyances filter

### Issue Details

Steps to reproduce:
1. On Poco F3 (Android 13, MIUI 14) with AdGuard enabled (Automatic Proxy - Routing Mode)
2. Open Settings, Google, Backup, Turn On Backup
3. The Backup sync stuck checking status of Apps, SMS and MMS messages, Call history, Device settings. The sync never ends. (screenshot annex)
4. Close the Google Backup (System App)
5. Disable AdGuard app
6. Open Settings, Google, Backup, Turn On Backup again
7. The Backup sync checking status of Apps, SMS and MMS messages, Call history, Device settings done instantaneously. None issue when are AdGuard disabled. (screenshot annex)

### Expected Behavior

The System Google Backup Sync works flawlessly when AdGuard enabled even if some exclusion filter for backup sync must be created (workaround). 

### Actual Behavior

System Google Backup Sync stuck when Adguard 4.1 (Betta) for Android (Enabled)

### Screenshots

<details><summary>Screenshots:</summary>

![1693717606790](https://github.com/AdguardTeam/AdguardForAndroid/assets/44372413/aa59614c-467a-4b6c-8715-9e53a14733d4)

![1693717606800](https://github.com/AdguardTeam/AdguardForAndroid/assets/44372413/c5b8be48-df25-4efd-9ecb-14436d9384e9)

![1693717606806](https://github.com/AdguardTeam/AdguardForAndroid/assets/44372413/ad464760-469a-4af2-87bb-ec5bed9800cb)

![1693717606811](https://github.com/AdguardTeam/AdguardForAndroid/assets/44372413/535c6738-47b8-4427-8de2-1d3353ad6ae3)

</details>

### Additional Information

Recently I did a clean flesh install of my Poco F3 (Xiaomi) and I'm running only a few apps as Google Drive, Google Authenticator, WhatsApp over Android 13 (MIUI 14) and all apps that are Google Backup Sync dependent stuck when accessing Cloud (upload/download).
I didn't noticed any blocking event in the AdGuard - Recent Activity (only some few blocked ads) that could justify this behaviour or identify the app that had been filtered/blocked by AdGuard and there is no obvious way to add in the allowlist filter.
That´s a huge issue once to do backup/restore/sync data it's mandatory to disable AdGuard.
How can we handle it? Let me know if other inputs are needed to fix it or even if you know some workaround.
Thanks in advance,

**Comment by Birraque** [comment]
Log File (Debug) and Screen Recorded when reproducing the bug already sent.
Thanks in advance and let me know if more inputs needed.
Regards,

**Comment by Birraque** [comment]
> @Birraque Hi! In order to investigate this case, we would like to take a look at the logs from your device. Could you collect them using the steps below?
> 
> 1. Enable debug logging:
>    _Settings -> General -> Advanced -> Logging level -> Debug_
> 2. Reproduce the issue, then remember the exact time when it happened
> 3. Collect logs:
>    _Settings -> General -> Advanced -> Export logs and system info_
> 4. Send this file to [devteam@adguard.com](mailto:devteam@adguard.com):
>    
>    * mention the `4957` number in the subject
>    * specify the exact time when the issue occurred

Did you receive the log file?

**Comment by Versty** [comment]
@Birraque Yes, we are examining your logs. 
If you try to reproduce this same issue using the Local VPN operating mode, would it be the case then?

**Comment by Birraque** [comment]
> @Birraque Yes, we are examining your logs. If you try to reproduce this same issue using the Local VPN operating mode, would it be the case then?

The Reported Bug only happens when Automatic Proxy (root Mode) selected.

![Screenshot_2023-10-06-01-39-40-144_com adguard android](https://github.com/AdguardTeam/AdguardForAndroid/assets/44372413/0bd19165-32b1-4de2-8399-81d53ee73a46)

**Comment by Versty** [comment]
@Birraque Hi! 
We managed to reproduce this issue on our side. We will find a way to fix this bug in the future versions. 

Please use Local VPN operating mode as a temporary solution.

**Comment by Birraque** [comment]
> @Birraque Hi! We managed to reproduce this issue on our side. We will find a way to fix this bug in the future versions.
> 
> Please use Local VPN operating mode as a temporary solution.

Not a workaround for me.
I rooted the device to use AdGuard.
I need the VPN for corporate use.

**Comment by Jrmewam** [comment]
I just created ticket #832674 through the app's bug reporting function.
Other apps are impacted, eg. Discord, 9gag, Spark...

**Comment by Birraque** [comment]
> I just created ticket #832674 through the app's bug reporting function. Other apps are impacted, eg. Discord, 9gag, Spark...

I totally agree with you.
Almost all apps and system been impacted when Proxy "root" selected.
The only functional workaround for me was too disable AdGuard for...
- Google Play Services (com.google.android.gms)
- All Bank apps
- All Backup apps
- All Cloud Storage Sync apps (Google Drive, OneDrive)
- All Authenticator apps (Google, Microsoft)
- All Email apps
- All Calendar apps
- All Contact apps
And still counting...
That's unacceptable AdGuard hadn't address it yet.

**Comment by Keyaku** [comment]
Automatic Proxy has been mostly working so far, except for Microsoft services or websites, where it fails outright until I deactivate protection.
None of the affected apps show the domains being blocked.

I'm guessing Automatic proxy is very complex to maintain.

**Comment by Versty** [comment]
@Birraque Can you confirm whether the Google Backup Sync issue still occurs in Automatic proxy mode?

**Comment by Birraque** [comment]
> [@Birraque](https://github.com/Birraque) Can you confirm whether the Google Backup Sync issue still occurs in Automatic proxy mode?

I gave up from Proxy mode long time ago. Never worked for me. 
I'm currently in AdGuard VPN and backup are working fine.

**Comment by Versty** [comment]
We have re-checked. Google Backup Sync is now working fine with the automatic proxy mode. If you are still experiencing the problem, please let us know and reopen this issue.

@Keyaku Hi! 
If you still observe any issues related to filtering in Microsoft services, please create [a separate issue](https://github.com/AdguardTeam/AdguardForAndroid/issues/new/choose) in this repository.
